utils/wesnoth-map-diff/src/paint.ts

```diff
@@ -22,71 +31,111 @@ const getTileImageCoordenates = (tileX: number, tileY: number) => {
     ? tileY * tileImageSize
     : tileY * tileImageSize - tileImageSize / 2
 
-  return [imageX, imageY]
+  return [axisMargin + imageX, axisMargin + imageY]
 }
 
-const producePainters = (output: Jimp, images: ImagesDict, leftPadding: number) => {
-  const paintTile = (x: number, y: number, baseCode: string) => {
-    const baseImage = images.tile[baseCode]
-    output.composite(baseImage, x + leftPadding, y)
-  }
-
-  // todo: we should use the defaultBase correctly
-  const paintMisc = (x: number, y: number, baseCode: string, miscCode: string) => {
-    if (!miscCode) {
-      return
-    }
-
-    const misctileImage = images.tile[`${baseCode}^${miscCode}`] || images.tile[`^${miscCode}`]
-    output.composite(misctileImage, x + leftPadding, y)
+const producePainters = (output: Jimp, images: ImagesGetter, leftPadding: number) => {
+  const paintTile = (x: number, y: number, baseCode: string, miscCode?: string) => {
+    const tileImage = images.getTile({ baseCode, miscCode })
+    output.composite(tileImage, x + leftPadding, y)
```

Comment:
There really should be a check here or in `getTile` and error output if the tile isn't actually found.